src/internal/operations/restore.go

```diff
@@ -228,6 +229,25 @@ func (op *RestoreOperation) do(
 		"restore_protected_resource_id", restoreToProtectedResource.ID(),
 		"restore_protected_resource_name", clues.Hide(restoreToProtectedResource.Name()))
 
+	// Check if the resource has the service enabled to be able to restore.
+	runnable, err := op.rc.IsServiceEnabled(
+		ctx,
+		op.Selectors.PathService(),
+		restoreToProtectedResource.ID())
+	if err != nil {
+		logger.CtxErr(ctx, err).Error("verifying restore is runnable")
```

Comment:
This is going to double-log the error when it gets caught again on line 171.  You can drop this log and let things bubble up.

Reply:
thanks! These are existing logs that I carried over during refactor. Will remove.